src/stack/Stack.java

```diff
@@ -37,20 +42,30 @@ public int pop() {
 	 */
 	public void clear() {
 
+
 	}
 
 	/**
 	 * Check if stack is empty
 	 * @return
 	 */
 	public boolean isEmpty() {
-		return false;
+		if (tos < 0) {
+			return true;
+		}else {
+			return false;
+		}
 	}
 
 	/**
 	 * Increase the stack capacity if there is not enough space to add additional items
 	 */
 	private void ensureCapacity() {
-
+		if (tos >= data.length) {
```

Comment:
This is private method and should work in some conditions. So, you don't need this **if statement** here.